### PR TITLE
feat(components/typeahead): align to guidelines

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

Our Typeahead component is a bit outdated : 
image

**What is the chosen solution to this problem?**

Align it to the new guidelines : http://guidelines.talend.com/document/92132#/ui-controls/typeahead

image

- A new `manageNavigation` prop has been added in order to optionally let the component manage the up/down nav (currently, every projects has to reimplement it).
- A new `hint` boolean can be set to `true` in the sections. The section will render has a... hint : 
image

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Typeahead/Typeahead.component.js [hunk 1]
```diff
@@ -35,11 +36,13 @@ function getItems(items, dataFeature) {
  *
  * <Typeahead {...props} />
  */
-function Typeahead({ onToggle, icon, position, docked, ...rest }) {
+function Typeahead({ onToggle, icon, position, docked, manageNavigation, ...rest }) {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Typeahead/Typeahead.component.js:39]: Could we not pass a flag but base this assertion with the presence of the onKeyDown prop ?

File: packages/components/src/Typeahead/Typeahead.component.js [hunk 2]
```diff
@@ -55,6 +58,34 @@ function Typeahead({ onToggle, icon, position, docked, ...rest }) {
 		);
 	}
 
+	const handleKeyDown = (event, data) => {
```
Comment (frassinier) [inline packages/components/src/Typeahead/Typeahead.component.js:61]: Does it throw Pendo events?

File: packages/components/src/Typeahead/Typeahead.component.renderers.js [hunk 3]
```diff
@@ -27,7 +30,7 @@ export function renderInputComponent(props) {
 	} = props;
 
 	const hasCaret = caret && !disabled && !readOnly;
-	const hasIcon = icon || hasCaret;
+	const hasIcon = icon;
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Typeahead/Typeahead.component.renderers.js:33]: You can remove this const, just use `icon`

File: packages/components/src/Typeahead/Typeahead.scss [hunk 4]
```diff
@@ -36,7 +37,7 @@ $tc-typeahead-items-zindex: 1035 !default;
 
 	.items-container,
 	.items-body {
-		max-height: 70vh;
+		max-height: 32vh;
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Typeahead/Typeahead.scss:40]: This seems to be 3.2rem in classic mode and 4.8rem with category in the guideline. hmm this doesn't feel right, we should have a single size. Can you check with the UX ?

File: packages/components/src/Typeahead/Typeahead.component.js [hunk 5]
```diff
@@ -96,7 +127,7 @@ function Typeahead({ onToggle, icon, position, docked, ...rest }) {
 			onChange: rest.onChange && (event => rest.onChange(event, { value: event.target.value })),
 			onFocus: rest.onFocus,
 			onClick: rest.onClick,
-			onKeyDown: rest.onKeyDown,
+			onKeyDown: manageNavigation ? handleKeyDown : rest.onKeyDown,
```
Comment (frassinier) [inline packages/components/src/Typeahead/Typeahead.component.js:130]: It's not enough to rely on onKeyDown props existence?

File: packages/components/src/Typeahead/Typeahead.stories.js [hunk 6]
```diff
@@ -39,10 +57,10 @@ const items = [
...
 	{
 		title: 'category 3',
-		icon: {
-			name: 'fa fa-asterisk',
-			title: 'icon',
-		},
+		// icon: {
+		// 	name: 'fa fa-asterisk',
+		// 	title: 'icon',
+		// },
```
Comment (jsomsanith-tlnd) [inline packages/components/src/Typeahead/Typeahead.stories.js:63]: Remove the commented code

File: packages/components/src/Typeahead/Typeahead.component.js (hunk 5 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.component.js:130]: I wanted to keep the opportunity to add additional behaviours to the handleKey but yes, you're right, it's more elegant like that. I push a fix, thanks

File: packages/components/src/Typeahead/Typeahead.component.js (hunk 1 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.component.js:39]: I wanted to keep the opportunity to add additional behaviours to the handleKey but yes, you're right, it's more elegant like that. I push a fix, thanks

File: packages/components/src/Typeahead/Typeahead.stories.js (hunk 6 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.stories.js:63]: oops

File: packages/components/src/Typeahead/Typeahead.component.renderers.js (hunk 3 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.component.renderers.js:33]: re-oops

File: packages/components/src/Typeahead/Typeahead.component.js (hunk 2 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.component.js:61]: definitely not

Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.component.js:61]: hm

File: packages/components/src/Typeahead/Typeahead.scss (hunk 4 above)
Comment (ncomont) [reply packages/components/src/Typeahead/Typeahead.scss:40]: This is about the items container, not the item itself. The usecase you're talking about is managed here : https://github.com/Talend/ui/pull/3114/files#diff-f73bbee543e731629b9cc27277a87ea17ccc17624cebe6e1f287e8f23a334a6bR191

**Review by frassinier — APPROVED** [review]

**Review by jsomsanith-tlnd — APPROVED** [review]

**Pull request merged**